packages/app-shell/src/components/AuthenticatedAppShellHeaderMenu.tsx

```diff
@@ -0,0 +1,123 @@
+import { forwardRef, useState } from 'react';
+import { boxPalette, mapSpacing } from '@ag.ds-next/react/core';
+import { Box, Flex } from '@ag.ds-next/react/box';
+import { Avatar } from '@ag.ds-next/react/avatar';
+import { Text } from '@ag.ds-next/react/text';
+import { ChevronDownIcon } from '@ag.ds-next/react/icon';
+import { useLinkComponent } from '@ag.ds-next/react/core';
+import { authenticatedAppShellHeaderHeight } from './utils';
+import { useAuthenticatedAppShellContext } from './AuthenticatedAppShellContext';
+import {
+	Menu,
+	MenuList,
+	MenuButton,
+	MenuItem,
+	MenuLink,
+} from '@reach/menu-button';
+
+export function AuthenticatedAppShellHeaderMenu() {
+	const { userMenu } = useAuthenticatedAppShellContext();
+	const Link = useLinkComponent();
+
+	const [refEl, setRefEl] = useState<HTMLButtonElement | null>(null);
+	const menuButtonWidth = refEl?.clientWidth;
+
+	return (
+		<Menu>
+			<AuthenticatedAppShellHeaderMenuButton ref={setRefEl} />
+			<Box as={MenuList} background="body" focus>
+				{userMenu.items.map((item, idx) => {
+					const { icon: Icon } = item;
+					if ('href' in item) {
+						return (
+							<MenuLink
+								key={idx}
+								as={Link}
+								href={item.href}
+								css={{ ...menuItemStyles, width: menuButtonWidth }}
+							>
+								{Icon ? <Icon /> : null}
+								{item.label}
+							</MenuLink>
+						);
+					}
+					return (
+						<MenuItem
+							key={idx}
+							onSelect={() => item.onClick()}
+							css={{ ...menuItemStyles, width: menuButtonWidth }}
```

Comment:
the `onClick` menu items don't get a pointer cursor

```suggestion
							css={{ ...menuItemStyles, width: menuButtonWidth, cursor: 'pointer' }}
```

Reply:
Done 👍